### ISSUE TITLE
Consider methods for operator overloading to be public

### ISSUE THREAD

**Issue by mthuurne** [PyCQA/pylint#1933]
### Steps to reproduce

Run PyLint on the following code:
```python
class C(object):

    def __getattribute__(self, key):
        return 1

    def __getitem__(self, key):
        return 2
```

### Current behavior

The following issue is reported:
```
R:  1, 0: Too few public methods (0/2) (too-few-public-methods)
```
### Expected behavior

Since the two methods are designed to be called from any code, I think it makes sense to consider them to be public.

### pylint --version output

```
No config file found, using default configuration
pylint-2.7 1.8.2, 
astroid 1.6.1
Python 2.7.13 (default, Jan 03 2017, 17:41:54) [GCC]
```

**Comment by PCManticore** [comment]
Fixed in master branch by @AWhetter !
